In [8]:
import pandas as pd
df=pd.read_excel(r"C:\Users\boomi\Downloads\superstore.xls")
df

,Row ID,Order ID,Order Date,Ship Date,Ship Mode,Customer ID,Customer Name,Segment,Country/Region,City,...,Postal Code,Region,Product ID,Category,Sub-Category,Product Name,Sales,Quantity,Discount,Profit
0,1,US-2023-103800,2023-01-03,2023-01-07,Standard Class,DP-13000,Darren Powers,Consumer,United States,Houston,...,77095,Central,OFF-PA-10000174,Office Supplies,Paper,"Message Book, Wirebound, Four 5 1/2"" X 4"" Form...",16.448,2,0.2,5.5512
1,2,US-2023-112326,2023-01-04,2023-01-08,Standard Class,PO-19195,Phillina Ober,Home Office,United States,Naperville,...,60540,Central,OFF-BI-10004094,Office Supplies,Binders,GBC Standard Plastic Binding Systems Combs,3.540,2,0.8,-5.4870
2,3,US-2023-112326,2023-01-04,2023-01-08,Standard Class,PO-19195,Phillina Ober,Home Office,United States,Naperville,...,60540,Central,OFF-LA-10003223,Office Supplies,Labels,Avery 508,11.784,3,0.2,4.2717
3,4,US-2023-112326,2023-01-04,2023-01-08,Standard Class,PO-19195,Phillina Ober,Home Office,United States,Naperville,...,60540,Central,OFF-ST-10002743,Office Supplies,Storage,SAFCO Boltless Steel Shelving,272.736,3,0.2,-64.7748
4,5,US-2023-141817,2023-01-05,2023-01-12,Standard Class,MB-18085,Mick Brown,Consumer,United States,Philadelphia,...,19143,East,OFF-AR-10003478,Office Supplies,Art,Avery Hi-Liter EverBold Pen Style Fluorescent ...,19.536,3,0.2,4.8840
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
10189,10190,US-2026-143259,2026-12-30,2027-01-03,Standard Class,PO-18865,Patrick O'Donnell,Consumer,United States,New York City,...,10009,East,OFF-BI-10003684,Office Supplies,Binders,Wilson Jones Legal Size Ring Binders,52.776,3,0.2,19.7910
10190,10191,US-2026-115427,2026-12-30,2027-01-03,Standard Class,EB-13975,Erica Bern,Corporate,United States,Fairfield,...,94533,West,OFF-BI-10004632,Office Supplies,Binders,GBC Binding covers,20.720,2,0.2,6.4750
10191,10192,US-2026-156720,2026-12-30,2027-01-03,Standard Class,JM-15580,Jill Matthias,Consumer,United States,Loveland,...,80538,West,OFF-FA-10003472,Office Supplies,Fasteners,Bagged Rubber Bands,3.024,3,0.2,-0.6048
10192,10193,US-2026-143259,2026-12-30,2027-01-03,Standard Class,PO-18865,Patrick O'Donnell,Consumer,United States,New York City,...,10009,East,TEC-PH-10004774,Technology,Phones,Gear Head AU3700S Headset,90.930,7,0.0,2.7279


In [2]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10194 entries, 0 to 10193
Data columns (total 21 columns):
 #   Column          Non-Null Count  Dtype         
---  ------          --------------  -----         
 0   Row ID          10194 non-null  int64         
 1   Order ID        10194 non-null  object        
 2   Order Date      10194 non-null  datetime64[ns]
 3   Ship Date       10194 non-null  datetime64[ns]
 4   Ship Mode       10194 non-null  object        
 5   Customer ID     10194 non-null  object        
 6   Customer Name   10194 non-null  object        
 7   Segment         10194 non-null  object        
 8   Country/Region  10194 non-null  object        
 9   City            10194 non-null  object        
 10  State/Province  10194 non-null  object        
 11  Postal Code     10194 non-null  object        
 12  Region          10194 non-null  object        
 13  Product ID      10194 non-null  object        
 14  Category        10194 non-null  object        
 15  Su

In [4]:
df.shape

(10194, 21)

In [5]:
total_sales = df['Sales'].sum()
total_profit = df['Profit'].sum()

print("Total Sales:", total_sales)
print("Total Profit:", total_profit)

Total Sales: 2326534.3543
Total Profit: 292296.8145999999


#Converting Total Sales and Profit to Indian Rupee Format

In [11]:
from babel.numbers import format_currency

total_sales = df['Sales'].sum()
total_profit = df['Profit'].sum()

print("Total Sales:", format_currency(round(total_sales), 'INR', locale='en_IN'))
print("Total Profit:", format_currency(round(total_profit), 'INR', locale='en_IN'))


Total Sales: ₹23,26,534.00
Total Profit: ₹2,92,297.00


In [6]:
df.groupby('Region')['Sales'].sum().sort_values(ascending=False)

Region
West       739813.6085
East       691828.1680
Central    503170.6728
South      391721.9050
Name: Sales, dtype: float64

In [7]:
df.groupby('Category')['Profit'].sum().sort_values(ascending=False)

Category
Technology         146543.3756
Office Supplies    126023.4434
Furniture           19729.9956
Name: Profit, dtype: float64

In [8]:
df.groupby('Sub-Category')['Profit'].sum().sort_values()

Sub-Category
Tables        -17753.2061
Bookcases      -3632.0736
Supplies       -1171.3945
Fasteners       2428.6358
Machines        3461.9769
Labels          5572.7780
Art             6653.1962
Envelopes       6988.0247
Furnishings    13891.7430
Appliances     18329.4844
Storage        21285.1115
Chairs         27223.5323
Binders        31426.1003
Paper          34511.5070
Accessories    41936.6357
Phones         45050.8265
Copiers        56093.9365
Name: Profit, dtype: float64

In [9]:
df['Month'] = df['Order Date'].dt.month
df.groupby('Month')['Sales'].sum()

Month
1      96069.5156
2      59751.2514
3     210672.0868
4     138098.7486
5     158765.0337
6     153459.9973
7     149001.7130
8     161821.9150
9     308883.6697
10    207126.2567
11    352665.9910
12    330218.1755
Name: Sales, dtype: float64

In [10]:
df[['Sales','Profit','Discount','Quantity']].corr()

,Sales,Profit,Discount,Quantity
Sales,1.000000,0.481460,-0.027756,0.198457
Profit,0.481460,1.000000,-0.218882,0.066145
Discount,-0.027756,-0.218882,1.000000,0.007475
Quantity,0.198457,0.066145,0.007475,1.000000


In [11]:
df['Month'] = df['Order Date'].dt.month_name()

In [12]:
monthly_sales = df.groupby('Month')['Sales'].sum()
monthly_sales

Month
April        138098.7486
August       161821.9150
December     330218.1755
February      59751.2514
January       96069.5156
July         149001.7130
June         153459.9973
March        210672.0868
May          158765.0337
November     352665.9910
October      207126.2567
September    308883.6697
Name: Sales, dtype: float64

In [13]:
monthly_sales.idxmax()

'November'

# Exploratory Data Analysis

1.The dataset contains 10,194 rows and 21 columns.
2.There are no missing values in the dataset.
3.Total Sales: ₹23,26,534.00
4.Total Profit: ₹2,92,297.00
5.The highest sales generating region is West.
6.The most profitable category is Technology.
7.The loss-making sub-categories are Tables and Bookcases.
8.Discount shows a negative correlation with profit.
9.The month with highest sales is November.